x/concentrated-liquidity/msg_server_test.go

```diff
@@ -137,18 +137,45 @@ func (suite *KeeperTestSuite) TestCollectFees_Events() {
 	testcases := map[string]struct {
 		upperTick                int64
 		lowerTick                int64
-		positionId               uint64
+		positionIds              []uint64
+		numPositionsToCreate     int
 		expectedCollectFeesEvent int
 		expectedMessageEvents    int
 		expectedError            error
 		errorFromValidateBasic   error
 	}{
-		"happy path": {
+		"single position ID": {
+			upperTick:                DefaultUpperTick,
+			lowerTick:                DefaultLowerTick,
+			positionIds:              []uint64{DefaultPositionId},
+			numPositionsToCreate:     1,
+			expectedCollectFeesEvent: 1,
+			expectedMessageEvents:    2, // 1 for collect fees, 1 for send message
+		},
+		"two position IDs": {
+			upperTick:                DefaultUpperTick,
+			lowerTick:                DefaultLowerTick,
+			positionIds:              []uint64{DefaultPositionId, DefaultPositionId + 1},
+			numPositionsToCreate:     2,
+			expectedCollectFeesEvent: 1,
+			expectedMessageEvents:    3, // 1 for collect fees, 2 for send messages
+		},
+		"three position IDs": {
 			upperTick:                DefaultUpperTick,
 			lowerTick:                DefaultLowerTick,
-			positionId:               DefaultPositionId,
+			positionIds:              []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+			numPositionsToCreate:     3,
 			expectedCollectFeesEvent: 1,
-			expectedMessageEvents:    2, // 1 for collect fees, 1 for message
+			expectedMessageEvents:    4, // 1 for collect fees, 3 for send messages
+		},
+		"error": {
+			upperTick:                DefaultUpperTick,
+			lowerTick:                DefaultLowerTick,
+			positionIds:              []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+			numPositionsToCreate:     2,
+			expectedCollectFeesEvent: 0,
+			expectedMessageEvents:    2, // 2 emitted for send messages
```

Comment:
Slightly questioning this. The first two bank sends "go through" and emit the send event, then the third one fails since the position doesn't exist. Despite this, the two bank send messages are still showing in the ctx. Is this expected behavior for events?

Reply:
yeah, I'm pretty sure these events would get dropped later somewhere in `runTx` logic of the baseapp